integration-tests/deployment/multiclient.go

```diff
@@ -39,53 +44,95 @@ type RPC struct {
 var _ OnchainClient = &MultiClient{}
 
 type MultiClient struct {
+	logger logger.Logger
 	*ethclient.Client
 	Backups     []*ethclient.Client
 	RetryConfig RetryConfig
 }
 
-func NewMultiClient(rpcs []RPC, opts ...func(client *MultiClient)) (*MultiClient, error) {
+func WithRetryConfig(attempts uint64, delay time.Duration) func(client *MultiClient) {
+	return func(client *MultiClient) {
+		client.RetryConfig = RetryConfig{
+			Attempts: attempts,
+			Delay:    delay,
+		}
+	}
+}
+
+func NewMultiClient(lggr logger.Logger, rpcs []RPC, opts ...func(client *MultiClient)) (*MultiClient, error) {
 	if len(rpcs) == 0 {
-		return nil, fmt.Errorf("No RPCs provided, need at least one")
+		return nil, fmt.Errorf("no RPCs provided, need at least one")
+	}
+	mc := &MultiClient{
+		logger: lggr,
 	}
-	var mc MultiClient
 	clients := make([]*ethclient.Client, 0, len(rpcs))
-	for _, rpc := range rpcs {
+	for i, rpc := range rpcs {
 		client, err := ethclient.Dial(rpc.WSURL)
 		if err != nil {
-			return nil, errors.Wrapf(err, "failed to dial %s", rpc.WSURL)
+			lggr.Warnf("failed to dial rpc %d ending in %s, moving to next one", i+1, rpc.WSURL[len(rpc.WSURL)-4:])
```

Comment:
`rpc.WSURL[len(rpc.WSURL)-4:])` could be just `/ws/` for a lot of them, but I guess that's better than having to count RPCs in the TOML or exposing more of the URL. Another option is to load in "name" from the TOML and use `rpc.Name`

Reply:
updated